File: daily_python/2138_Divide_a_String_Into_Groups_of_Size_k.py
import sys
import os
from typing import List

class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        n = len(s)
        ans = []
        s = s + fill * (k - n % k)
        for idx in range(0, n, k):
            ans.append(s[idx:idx + k])
        return ans
    
def readData(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            s = lines[0].split('=')[1].strip()[1:-2]
            k = int(lines[1].split('=')[1][:-1])
            fill = lines[2].split('=')[1].strip()[1:-2]
            dataset.append((s, k, fill))
    return dataset

if __name__ == '__main__':
    if (len(sys.argv) == 1):
        path = os.path.basename(__file__)
        filename = os.path.splitext(path)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readData(filename)
    results = []
    solution = Solution()
    for s, k, fill in dataset:
        results.append(solution.divideString(s, k, fill))
    for index, result in enumerate(results):
        print(f'dataset {index+1} : {result}')

File: daily/2025/2138_Divide_a_String_Into_Groups_of_Size_k.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        int n = s.length();
        vector<string> ans;
        s = s + string(k - n % k, fill);
        for (int i = 0; i < n; i += k){
            ans.push_back(s.substr(i, k));
        }
        return ans;
    }
};

int main(){
    string s = "abcdefghi";
    int k = 3;
    char fill = 'x';
    // string s = "abcdefghij";
    // int k = 3;
    // char fill = 'x';
    Solution S;
    vector<string> ans = S.divideString(s, k, fill);
    for (string i : ans) cout << i << " ";
    cout << endl;
    return 0;
}